JoonasHeinonen/escape-from-zordoom-prison
# feature/better-sky: Imported new sky images, added new tiles.

### [Better sky](https://trello.com/c/H4YC9FuV/51-better-sky)

![image](https://user-images.githubusercontent.com/33086486/209875332-02054aff-c97f-4fdf-bde5-d718423246c1.png)

**Motive:**

The sky in its original state in the `test_level.tscn` was not adequate, as it was just flat, dark-blue sky.

**Results:**

Managed to add 5 HDR panorama images, which can be used on designing sky for each level. Discovered methods on how to make both procedural and panorama sky. As an addition to this, I needed to fix separately old materials for models as they reflected the light generated by the `WorldEnvironment` node. As an addition to this, the original sun was too much light generated along with the light generated by the `WorldEnvironment` node. 

As an addition to this, added some new tiles to the `MeshLibrary`.

**How to test:**
 - Change the image for the `WorldEnvironment` node on the `test_level.tscn`.
 - See how the image affects the light generated to the scene.

**Conclusions:**

From now on, it would be advisable to check each material for a 3D model. Also, we should always rely on HDR panorama images we can use when designing a level.

**Sources cited:**
 - [Open World Environment Lighting Godot Engine 3D Tutorial](https://www.youtube.com/watch?v=8kwnCxK8Vc8&ab_channel=TokisanGames)
